website/source/guides/secret-mgmt/versioned-kv.html.md

```diff
@@ -0,0 +1,950 @@
+ "wrap_info": null,
+ "warnings": null,
+ "auth": null
+}
+```
+
+To read the **metadata** of `secret/customer/acme`:
+
+```plaintext
+$ curl --header "X-Vault-Token: ..." \
+       http://127.0.0.1:8200/v1/secret/metadata/customer/acme | jq
+{
+ "request_id": "34708262-59cd-9a94-247f-3b1db0909050",
+ "lease_id": "",
+ "renewable": false,
+ "lease_duration": 0,
+ "data": {
+   "created_time": "2018-04-14T00:05:47.115378933Z",
+   "current_version": 2,
+   "max_versions": 0,
+   "oldest_version": 0,
+   "updated_time": "2018-04-14T00:13:35.296018431Z",
+   "versions": {
+     "1": {
+       "created_time": "2018-04-14T00:05:47.115378933Z",
+       "deletion_time": "",
+       "destroyed": false
+     },
+     "2": {
+       "created_time": "2018-04-14T00:13:35.296018431Z",
+       "deletion_time": "",
+       "destroyed": false
+     }
+   }
+ },
+ "wrap_info": null,
+ "warnings": null,
+ "auth": null
+}
+```
+
+
+### <a name="step4"></a>Step 4: Specify the number of versions to keep
+
+By default, the `kv-v2` secret engine keeps up to 10 versions.  Let's limit the
+maximum number of versions to keep to be 4.
+
+#### CLI command
+
+To set the `secret/` to keep up to 4 versions:
+
+```shell
+$ vault write secret/config max_versions=4
+Success! Data written to: secret/config
+
+# View the configuration settings
+$ vault read secret/config
+Key             Value
+---             -----
+cas_required    false
+max_versions    4
+```
+
+Alternatively, to limit the number of versions only on the
+**`secret/customer/acme`** path rather than the entire `secret/` engine:
+
+```plaintext
+$ vault kv metadata put -max-versions=4 secret/customer/acme
+```
+
+Overwrite the data a few more times to see what happens to the data.
+
+```plaintext
+$ vault kv metadata get secret/customer/acme
+======= Metadata =======
+Key                Value
+---                -----
+created_time       2018-04-14T00:42:25.677078177Z
+current_version    6
+max_versions       0
+oldest_version     3
+updated_time       2018-04-16T00:17:23.930473344Z
+
+====== Version 3 ======
+Key              Value
+---              -----
+created_time     2018-04-16T00:15:59.880368849Z
+deletion_time    n/a
+destroyed        false
+
+====== Version 4 ======
+Key              Value
+---              -----
+created_time     2018-04-16T00:16:18.941331243Z
+deletion_time    n/a
+destroyed        false
+
+====== Version 5 ======
+Key              Value
+---              -----
+created_time     2018-04-16T00:16:34.407951572Z
+deletion_time    n/a
+destroyed        false
+
+====== Version 6 ======
+Key              Value
+---              -----
+created_time     2018-04-16T00:17:23.930473344Z
+deletion_time    n/a
+destroyed        false
+```
+
+In this example, the current version is 6. Notice that version 1 and 2 does not
+show up in the metadata. Because the kv secret engine is configured to keep only
+4 versions, the oldest two versions are permanently deleted that you won't be
+able to read it.
```

Comment:
the oldest two versions are permanently deleted and you won't be able to read them